src/core/client/admin/components/UserRole/SiteModeratorModal.tsx

```diff
@@ -71,50 +73,56 @@ const SiteModeratorModal: FunctionComponent<Props> = ({
-                    <ModalBodyText>
-                      Site moderators are permitted to make moderation decisions
-                      and issue suspensions on the sites they are assigned.
-                    </ModalBodyText>
-                  </Localized>
-                  <SiteModeratorModalSiteFieldContainer query={query} />
-                  <Flex justifyContent="flex-end" itemGutter="half">
-                    <Localized id="community-siteModeratorModal-cancel">
-                      <Button variant="flat" onClick={onCancel}>
-                        Cancel
-                      </Button>
+            {({ handleSubmit, submitError, submitting, values }) => {
+              return (
+                <form onSubmit={handleSubmit}>
+                  <HorizontalGutter spacing={3}>
+                    <Localized
+                      id="community-siteModeratorModal-assignSites"
+                      strong={<ModalHeaderUsername />}
+                      $username={username || notAvailableTranslation}
+                    >
+                      <ModalHeader>
+                        Assign sites for{" "}
+                        <ModalHeaderUsername>{username}</ModalHeaderUsername>
+                      </ModalHeader>
                     </Localized>
-                    <Localized id="community-siteModeratorModal-assign">
-                      <Button
-                        type="submit"
-                        disabled={submitting}
-                        ref={lastFocusableRef}
-                      >
-                        Assign
-                      </Button>
+                    {submitError && (
+                      <CallOut color="error" fullWidth>
+                        {submitError}
+                      </CallOut>
+                    )}
+                    <Localized id="community-siteModeratorModal-assignSitesDescription">
+                      <ModalBodyText>
+                        Site moderators are permitted to make moderation
+                        decisions and issue suspensions on the sites they are
+                        assigned.
+                      </ModalBodyText>
                     </Localized>
-                  </Flex>
-                </HorizontalGutter>
-              </form>
-            )}
+                    <SiteModeratorModalSites
+                      selectedSiteIDs={selectedSiteIDs}
+                    />
+                    <Flex justifyContent="flex-end" itemGutter="half">
+                      <Localized id="community-siteModeratorModal-cancel">
+                        <Button variant="flat" onClick={onCancel}>
+                          Cancel
+                        </Button>
+                      </Localized>
+                      <Localized id="community-siteModeratorModal-assign">
+                        <Button
+                          type="submit"
+                          disabled={submitting || values.siteIDs.length === 0}
```

Comment:
The main changes to this file are adding this `disabled` check that at least one site has been selected, as well as adding `data-testids` for the modal and submit button. And removing the `query` prop.